File: 백준/Silver/10828. 스택/스택.swift
import Foundation

struct Stack {
    var stack: [Int] = []
    
    mutating func push(x: Int) {
        stack.append(x)
    }
    
    mutating func pop() -> Int {
        return stack.isEmpty ? -1 : stack.removeLast()
    }
    
    func size() -> Int {
        return stack.count
    }
    
    func empty() -> Int {
        return stack.isEmpty ? 1 : 0
    }
    
    func top() -> Int {
        return stack.last ?? -1
    }
}

let n = Int(readLine()!)!

var stack = Stack()

for _ in 0..<n {
    let line = readLine()!.split(separator: " ")
    switch line[0] {
    case "push" :
        stack.push(x: Int(line[1])!)
    case "pop" :
        print(stack.pop())
    case "size" :
        print(stack.size())
    case "empty" :
        print(stack.empty())
    case "top" :
        print(stack.top())
    default:
        break
    }
}

File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <stack>

using namespace std;

int main()
{
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    
    stack<int> s;
    
    int n;
    cin >> n;
    
    while(n--) {
        string input;
        cin >> input;
        if (input=="push") {
            int add;
            cin >> add;
            s.push(add);
            
        } else if (input=="top") {
            if (s.empty())
                cout << -1 << "\n";
            else
                cout << s.top() << "\n";
            
        } else if (input=="pop") {
            if (s.empty()) {
                cout << -1 << "\n";
            }
            else {
                cout << s.top() << "\n";
                s.pop();
            }
        } else if (input=="empty") {
            if (s.empty()) {
                cout << 1 << "\n";
            } else {
                cout << 0 << " ";
            }
        } else if (input == "size") {
            cout << s.size() << "\n";
        }
    }
}
